### PR TITLE
Global Styles: Add back the global styles logic that forces the solid border when color or width applied.

### PR DESCRIPTION
**Pull request by youknowriad**
Fixes a regression introduced int https://github.com/WordPress/gutenberg/pull/49428
See https://github.com/WordPress/gutenberg/pull/49428#issuecomment-1541441515

## What?

In Global Styles, we have some logic in place that forces the border style to be "solid" when a color/width is applied. This PR just restores that logic that was removed inadvertently. 

## Testing Instructions

1- Open the "button" global styles panel
2- Type 1px in the "border width" input
3- Verify that the "solid" style is applied on the button

### CODE REVIEW

File: packages/edit-site/src/components/global-styles/screen-block.js [hunk 1]
```diff
@@ -152,15 +216,15 @@ function ScreenBlock( { name, variation } ) {
...
+					onChange={ onChangeBorders }
 					settings={ settings }
 				/>
 			) }
 			{ hasEffectsPanel && (
 				<StylesEffectsPanel
 					inheritedValue={ inheritedStyleWithLayout }
 					value={ styleWithLayout }
-					onChange={ onChangeDimensions }
+					onChange={ setStyle }
```
Comment (youknowriad) [inline packages/edit-site/src/components/global-styles/screen-block.js:234]: This is an unrelated change but it's a small optim/typo fix (same for the next change)

**Review by aaronrobertshaw — COMMENTED** [review]
Thanks for the quick fix @youknowriad 🚀 

It looks like we'll need another small tweak to match the current code in the [global styles border panel](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/global-styles/border-panel.js#L70-L104).

Also, can we now remove the [global styles border panel](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/global-styles/border-panel.js) entirely?

I'm out of time today but will give this a full test tomorrow and can make any further updates needed then if you'd like.

**Comment by youknowriad** [comment]
Updated. Indeed, the border panel seems like dead code 👍

**Review by aaronrobertshaw — APPROVED** [review]
LGTM 🚢 

✅ Border controls work as expected in both block editor and global styles
✅ Code changes look good including old global styles border panel removal
✅ Border radius is maintained when adjusting other global styles border properties

The failing unit tests look related to the Fonts API that I believe was reverted overnight. I'll rebase the PR and see if we can't get them passing.

Thanks again for the quick work on this.

**Pull request merged**